.github/scripts/generate_ci_workflows.py

```diff
@@ -1,123 +1,106 @@
 #!/usr/bin/env python3
 
+from dataclasses import asdict, dataclass
 from pathlib import Path
-from typing import Any, Dict, Optional
 
 import jinja2
 from typing_extensions import Literal
 
-DOCKER_REGISTRY = "308535385114.dkr.ecr.us-east-1.amazonaws.com"
+YamlShellBool = Literal["''", 1]
+Arch = Literal["windows", "linux"]
 
+DOCKER_REGISTRY = "308535385114.dkr.ecr.us-east-1.amazonaws.com"
 GITHUB_DIR = Path(__file__).resolve().parent.parent
 
-
-# it would be nice to statically specify that build_environment must be
-# present, but currently Python has no easy way to do that
-# https://github.com/python/mypy/issues/4617
-PyTorchWorkflow = Dict[str, Any]
-
-YamlShellBool = Literal["''", 1]
-
 WINDOWS_CPU_TEST_RUNNER = "windows.4xlarge"
 WINDOWS_CUDA_TEST_RUNNER = "windows.8xlarge.nvidia.gpu"
+WINDOWS_RUNNERS = set([
+    WINDOWS_CPU_TEST_RUNNER,
+    WINDOWS_CUDA_TEST_RUNNER,
+])
```

Comment:
if you want, you can just use a set literal (doesn't matter though)
```suggestion
WINDOWS_RUNNERS = {
    WINDOWS_CPU_TEST_RUNNER,
    WINDOWS_CUDA_TEST_RUNNER,
}
```